js/filters.js

```diff
@@ -114,7 +120,11 @@ weechat.filter('DOMfilter', ['$filter', '$sce', function($filter, $sce) {
             if (node === undefined || node === null) return;
             node = node.firstChild;
             while (node) {
-                var nextNode = process(node);
+                var nextNode;
+                // do not recurse inside links if the filter would create a nested link
+                if (!(createsAnchor && node.tagName == 'A')) {
```

Comment:
I'm sure you know it works correctly, but others may not. Loose equality is so strange in JS that I prefer to avoid it completely. And that check above could be even shorter using `(!node)`